nemo/collections/nlp/nm/data_layers/qa_squad_datalayer.py

```diff
@@ -59,9 +59,9 @@ def output_ports(self):
             "input_ids": NeuralType(('B', 'T'), ChannelType()),
             "input_type_ids": NeuralType(('B', 'T'), ChannelType()),
             "input_mask": NeuralType(('B', 'T'), ChannelType()),
-            "start_positions": NeuralType(tuple('B'), ChannelType()),
-            "end_positions": NeuralType(tuple('B'), ChannelType()),
             "unique_ids": NeuralType(tuple('B'), ChannelType()),
+            "start_positions": NeuralType(tuple('B'), ChannelType(), optional=True),
+            "end_positions": NeuralType(tuple('B'), ChannelType(), optional=True),
```

Comment:
@okuchaiev What exactly does optional mean? How does it work if I only return end_positions but not start_positions?

Reply:
start_positions, end_positions are the "labels", they are optional to make the data_layer more general for inference cases where i do not have the groundtruth.

Reply:
That makes sense, I just didn't know we had this option, and want to know how it works.